lock-api/src/test/java/com/palantir/lock/client/IdentifiedLockRequestTest.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.guava.GuavaModule;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+
+public class IdentifiedLockRequestTest {
+    private static final String SERIALIZED_LOCK_REQUEST = "{"
+            + "\"requestId\":\"885afd9c-de62-44ff-a517-5db14b71bfaa\","
+            + "\"lockDescriptors\":[{\"bytes\":\"Zm9v\"}],"
+            + "\"acquireTimeoutMs\":123,"
+            + "\"clientDescription\":\"Thread: main\"}";
+
+    private static final ObjectMapper mapper = new ObjectMapper()
```

Comment:
you should use the mapper you expect to be used, otherwise this test doesn't really mean anything. Also how much do you care about serde? Is there some backcompat concern that you're trying to uphold? If so, I'd adjust the test name to indicate that.

Reply:
This is a backcompat test I think (specifically, that an old `LockRequest` serializes to the new `IdentifiedLockRequest`)

Reply:
this is a backcompat test; I can rename this to make it more clear.
On object mapper part, I think we actually create object mappers and register modules on server and client side separately - there isn't one single ObjectMapper template used everywhere. This may be something we want to fix though (probably in a separate PR)

Reply:
if so, then I'd recommend serialising with the client object mapper we use, and then deserialising with the server object mapper we use, as this object mapper is quite far from what we use and not representative if we're doing a back compat test.

Reply:
problem is we don't have a well defined client and server object mappers. It is being created like this: https://github.com/palantir/atlasdb/blob/develop/atlasdb-feign/src/main/java/com/palantir/atlasdb/http/AtlasDbFeignTargetFactory.java#L53 (so currently there is no way to ensure we are using the same object mapper in the tests here)

I still think this is something we need to refactor though.